packages/backend-common/src/reading/UrlReaderMux.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { UrlReader } from './types';
+
+/**
+ * A UrlReader implementation that selects from a set of UrlReaders
+ * based on the host in the URL.
+ */
+export class UrlReaderMux implements UrlReader {
```

Comment:
Have named these consistently as the plural of the thing - UrlReaders

Reply:
You mean just call it plain `UrlReaders`? Thinking that this is API that's public but somewhat hidden in that there's a config layer that most ppl would use

Reply:
UrlReaders.fromConfig

Makes sense

Reply:
Yeah was thinking that's gonna exist, but that it will use this under the hood. Didn't wanna tie this implementation to other UrlReader implementations. Can see how it all fits together once there are more pieces in place though